core/dmux.go

```diff
@@ -311,7 +378,172 @@ func simpleSetup(size, qsize int, sink Sink) ([]chan interface{}, *sync.WaitGrou
+						log.Printf("error in serde of SidelineMeta")
+						continue
+					}
+					val := channelObject.Msg.(source.KafkaMsg)
+					log.Printf("Inside sideline channel for partition %d offset %d", val.GetRawMsg().Partition, val.GetRawMsg().Offset)
+					kafkaSidelineMessage := sideline_models.SidelineMessage{
+						GroupId:           string(val.GetRawMsg().Key),
+						Partition:         val.GetRawMsg().Partition,
+						EntityId:          string(val.GetRawMsg().Key) + sideline.ConsumerGroupName + sideline.ClusterName,
+						Offset:            val.GetRawMsg().Offset,
+						ConsumerGroupName: sideline.ConsumerGroupName,
+						ClusterName:       sideline.ClusterName,
+						Message:           val.GetRawMsg().Value,
+						Version:           channelObject.Version,
+						ConnectionType:    sideline.ConnectionType,
+						SidelineMeta:      sidelineMetaByteArray,
+					}
+
+					sidelineByteArray, err := json.Marshal(kafkaSidelineMessage)
+					if err != nil {
+						log.Printf("error in serde of kafkaSidelineMessage")
+						continue
+					}
+					sidelineMessageResponse := sidelineImpl.(sideline_models.CheckMessageSideline).SidelineMessage(sidelineByteArray)
+					if !sidelineMessageResponse.Success {
+						var check sideline_models.CheckMessageSidelineResponse
+						log.Printf(sidelineMessageResponse.ErrorMessage)
+						if sidelineMessageResponse.ConcurrentModificationError {
+							checkSidelineMessage := sideline_models.SidelineMessage{
+								GroupId:           string(val.GetRawMsg().Key),
+								Partition:         val.GetRawMsg().Partition,
+								EntityId:          string(val.GetRawMsg().Key) + sideline.ConsumerGroupName + sideline.ClusterName,
+								Offset:            val.GetRawMsg().Offset,
+								ConsumerGroupName: sideline.ConsumerGroupName,
+								ClusterName:       sideline.ClusterName,
+								Message:           val.GetRawMsg().Value,
+								ConnectionType:    sideline.ConnectionType,
+							}
+							checkSidelineMessageBytes, checkSidelineMessageErr := json.Marshal(checkSidelineMessage)
+							if checkSidelineMessageErr != nil {
+								log.Printf("error in serde of checkSidelineMessage")
+								errors.New("error in serde of checkSidelineMessage")
+								continue
+							}
+							checkBytes, checkErr := sidelineImpl.(sideline_models.CheckMessageSideline).
+								CheckMessageSideline(checkSidelineMessageBytes)
+							err := json.Unmarshal(checkBytes, &check)
+							if err != nil {
+								errors.New("error in serde of CheckMessageSidelineResponse")
+								continue
+							}
+							if checkErr != nil {
+								continue
+							}
+							channelObject.Version = check.Version
+							continue
+						}
+						continue
+					}
+					break
+				}
+			}
+		}(i)
+	}
+
+	for i := 0; i < size; i++ {
+		ch[i] = make(chan interface{}, qsize)
+		go func(index int) {
+			for msg := range ch[index] {
+				val := msg.(source.KafkaMsg)
+				var check sideline_models.CheckMessageSidelineResponse
+				for {
```

Comment:
Same. here, this needs to be modelled properly to handle retry, otherwise can cause runtime issues